flow/cmd/validate_mirror.go

```diff
@@ -61,24 +64,25 @@ func (h *FlowRequestHandler) ValidateCDCMirror(
 	)
 	if err != nil {
 		if errors.Is(err, errors.ErrUnsupported) {
-			return nil, errors.New("connector is not a supported source type")
+			return nil, exceptions.NewUnimplementedApiError("connector is not a supported source type")
 		}
-		return nil, fmt.Errorf("failed to create source connector: %w", err)
+		return nil, exceptions.NewFailedPreconditionApiError(fmt.Sprintf("failed to create source connector: %s", err))
 	}
 	defer connectors.CloseConnector(ctx, srcConn)
 
 	if err := srcConn.ValidateMirrorSource(ctx, req.ConnectionConfigs); err != nil {
-		return nil, fmt.Errorf("failed to validate source connector %s: %w", req.ConnectionConfigs.SourceName, err)
+		return nil, exceptions.NewFailedPreconditionApiError(
+			fmt.Sprintf("failed to validate source connector %s: %v", req.ConnectionConfigs.SourceName, err))
 	}
 
 	dstConn, err := connectors.GetByNameAs[connectors.MirrorDestinationValidationConnector](
 		ctx, req.ConnectionConfigs.Env, h.pool, req.ConnectionConfigs.DestinationName,
 	)
 	if err != nil {
 		if errors.Is(err, errors.ErrUnsupported) {
-			return &protos.ValidateCDCMirrorResponse{}, nil
+			return nil, exceptions.NewUnimplementedApiError("connector is not a supported destination type")
```

Comment:
@serprex this is causing tests like ES to fail

Reply:
I am reverting the change specifically for this place, but if we don't care about this, we should remove the check if possible